packages/apollo-server-core/src/ApolloServer.ts

```diff
@@ -419,6 +420,27 @@ export class ApolloServerBase {
         } => typeof maybeTaggedServerListener.serverListener === 'object',
       );
 
+      taggedServerListeners.forEach(
+        ({ serverListener: { schemaDidLoadOrUpdate } }) => {
+          if (schemaDidLoadOrUpdate) {
+            try {
+              schemaManager.onSchemaLoadOrUpdate(schemaDidLoadOrUpdate);
+            } catch (_) {
```

Comment:
I'm not sure I understand this try/catch.  schemaManager.onSchemaLoadOrUpdate can throw for multiple reasons, and one of those reasons already talks about gateway upgrades?

Reply:
The idea here was that:
- The error message thrown by `SchemaManager.onSchemaLoadOrUpdate()` on old gateways doesn't actually mention the hook name `schemaDidLoadOrUpdate`, since `SchemaManager.onSchemaLoadOrUpdate()` may in the future be called in other locations. So the try/catch is at the callsite to change the error message to mention the hook name.
- As for the `catch (_)`, I believe `SchemaManager.onSchemaLoadOrUpdate()` right now will only throw for that one specific case. This is admittedly fragile though, so I've switched to using a specific error subclass, and translating the error message just for that subclass (rethrowing otherwise).

I'm also down for any other approaches here you have in mind.